math/src/elliptic_curve/short_weierstrass/curves/bls12_381/twist.rs

```diff
@@ -46,7 +47,57 @@ impl IsShortWeierstrass for BLS12381TwistCurve {
     }
 }
 
+impl BLS12381TwistCurve {
+    pub fn is_on_curve(point: &<Self as IsEllipticCurve>::PointRepresentation) -> bool {
+        if point.z() == &FieldElement::zero() {
+            true
+        } else {
+            let point = point.to_affine();
+            Self::defining_equation(&point.x(), &point.y()) == FieldElement::zero()
+        }
+    }
+}
+
 impl ShortWeierstrassProjectivePoint<BLS12381TwistCurve> {
+    /// Returns the "Untwist-Frobenius-Twist" endomorphism
+    pub fn psi(&self) -> Self {
```

Comment:
Please add a test that checks that this endomorphism has minimal polynomial $X^2 - tX + q$. Maybe checking that on a few random points.